### PR TITLE
Rename multiplex to libp2p-mplex

### LINKED ISSUE #151: Rename the libraries so that the naming scheme is correct
Directories should simply be `identify`, `kad`, and so on, instead of `libp2p-identify`, `libp2p-kad`, etc.
The crate names, however, should have the `libp2p-` prefix.

### PR DESCRIPTION
**Pull request by tomaka**
Close #151

### CODE REVIEW

File: example/Cargo.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@ bytes = "0.4"
 env_logger = "0.5.4"
 futures = "0.1"
 multiaddr = "0.2"
-multiplex = { path = "../multiplex-rs" }
+libp2p-multiplex = { path = "../multiplex" }
```
Comment (diasdavid) [inline example/Cargo.toml]: actually, rename to libp2p-mplex. See the new spec https://github.com/libp2p/mplex

**Review by daviddias — CHANGES_REQUESTED** [review]

**Comment by Vurich** [comment]
I don't think `multiplex` has anything to do with `libp2p` necessarily. It's just a Rust implementation of the JS `multiplex` library

**Comment by tomaka** [comment]
It's tied to the other parts of libp2p though. For example it implements the `ConnectionReuse` trait.

**Comment by tomaka** [comment]
@diasdavid I renamed this to `mplex`, could you accept the changes? :)

**Comment by tomaka** [comment]
I don't understand why the master branch passes CI, meanwhile this PR doesn't because the version of Rust is too old.

**Review by daviddias — APPROVED** [review]

**Comment by diasdavid** [comment]
@VictorBjelkholm wanna help the rust team running their tests on Jenkins as well?

**Comment by VictorBjelkholm** [comment]
@diasdavid Sure, I'd be happy to help! Created issue for tracking it here: https://github.com/ipfs/jenkins-libs/issues/11

**Comment by tomaka** [comment]
Why is Jenkins needed, however? I was in fact thinking of ditching travis so that we only have to deal with one CI infrastructure.

**Comment by VictorBjelkholm** [comment]
@tomaka we're using jenkins across all ipfs, libp2p, multiformats, ipld and ipfs-shipyard. Basically the main features at this point would be faster builds, easier maintenance of pipelines, automatically add projects to jenkins by just adding a file in `ci/Jenkinsfile` and it'll automatically pick up, test results nicely highlighted ([example](https://ci.ipfs.team/blue/organizations/jenkins/IPFS%2Fjs-ipfs/detail/master/88/tests)), cross-platform builds (win, mac, linux and more to come in the future) in one CI service and some other minor benefits. In general, it's pretty good.

**Pull request merged**